completions/zsh/brew-bundle

```diff
@@ -0,0 +1,131 @@
+#compdef brew
+#autoload
+
+__brew_bundle_commands() {
+  local -a commands
+  commands=(
+    '-h:Show help'
+    '--help:Show help'
+    'install:Install and upgrade (by default) all dependencies from the Brewfile'
+    'dump:Write all installed casks/formulae/images/taps into a Brewfile in the current directory'
+    'cleanup:Uninstall all dependencies not listed from the Brewfile'
+    'check:Check if all dependencies are installed from the Brewfile'
+    'list:List all dependencies present in the Brewfile'
+    'exec:Run an external command in an isolated build environment based on the Brewfile dependencies'
+  )
+  _describe -t commands 'commands' commands
+}
+
+ __brew_bundle_expand_alias()
+ {
+   local command_or_alias="$1"
+   local -A aliases
+   aliases=(
+     setup install
+     set install
+     inst install
+     ins install
+     i install
+     du dump
+     d dump
+     clean cleanup
+     cl cleanup
+     rm cleanup
+     ch check
+     ls list
+     l list
+     ex exec
+     e exec
+   )
+   command="${aliases[$command_or_alias]:-$command_or_alias}"
+   print "${command}"
+ }
+
+_brew_bundle_install() {
+  _arguments \
+    '--file[read the Brewfile from this location. Use --file=- to pipe to stdin/stdout]' \
```

Comment:
I wonder if there's any way to rely on any of the Homebrew/brew code to autogenerate some/all of this or at least have CI to ensure it's kept in sync?

Reply:
I didn't research that :c
I know there are some tools to automate this script creation in golang, but not sure for ruby

Reply:
No worries, I'll try to help! Check out:

https://github.com/Homebrew/brew/blob/a56a2707075a10827b4c49b3cc51cae7bb94c2d6/completions/zsh/_brew#L4-L6

and

https://github.com/Homebrew/brew/blob/a56a2707075a10827b4c49b3cc51cae7bb94c2d6/Library/Homebrew/completions.rb

which you could maybe use for auto-generation?

Don't worry if you can't handle the e.g. GitHub Actions side: I can help with all that.

Reply:
Well, i'm not good at ruby (actually, it's my first try), but after some research of given code, i think i can make similar for `brew bundle`.
It requires some refactoring and i think, i would make it, if may

Reply:
@Tony-Sol just wanted to check how things are going and what you're thinking! Thanks again for the PR!

Reply:
@MikeMcQuaid sorry, i've been unlucky with free time lately :c

i remember about this PR, but can't say, when could continue work on it :c

Reply:
@Tony-Sol No worries! Just saw the rebase and wanted to check!